docs/user-guide/databases.rst

```diff
@@ -0,0 +1,208 @@
+#############
+
+Felis can create a database from a schema using the command line tool or the Python API.
+This will emit DDL which can either be executed automatically or written to a file for later use.
+
+Using the Command Line Tool
+===========================
+
+The ``felis create`` command can be used to create a database from the command line.
+For more information on the command line options for the ``create`` command, see the
+`command line documentation <cli.html#felis-create>`_.
+
+Dry Run Mode
+------------
+
+The dry run mode can be used to generate SQL without actually creating the database.
+This can be useful for inspecting the SQL that will be generated before running it.
+
+There are several different ways to run in dry run mode, one being to simply omit the ``--engine-url`` option:
+
+.. code-block:: bash
+
+    felis create schema.yaml
+
+This will by default instantiate the database to an in-memory (transient) SQLite database.
+Running such a command can be useful for finding errors in the schema file before attempting to create a
+persistent database.
+
+A specific database dialect such as MySQL can be selected in dry run mode using the ``--engine-url`` option:
+
+.. code-block:: bash
+
+    felis create --engine-url mysql:// schema.yaml
+
+This will generate the SQL for a MySQL database but will not actually create it.
+
+The dry run mode may also be explicitly enabled using the ``--dry-run`` option:
+
+.. code-block:: bash
+
+    felis create --dry-run --engine-url mysql+mysqlconnector://username:password@localhost schema.yaml schema.yaml
+
+The URL in this case will be used to determine the database dialect and generate the appropriate SQL but will
+otherwise be ignored.
+
+The generated SQL may also be saved to a file using the ``--output-file`` option:
+
+.. code-block:: bash
+
+    felis create --engine-url mysql:// schema.yaml --output-file schema.sql
+
+This SQL file could then be used to create the database at a later time using a tool such as the MySQL client.
+
+Creating a Persistent Database
+------------------------------
+
+In order to create a persistent database, the ``--engine-url`` must be set to a valid database URL.
+The URL format follows `SQLAlchemy engine conventions <https://docs.sqlalchemy.org/en/20/core/engines.html>`_:
+``dialect+driver://username:password@host:port/database``.
+
+Each of the parameters is explained below:
+
+- ``dialect``: The name of the database backend, such as ``sqlite``, ``mysql``, or ``postgresql``.
+- ``driver``: The name of the DBAPI to use, such as ``mysqlconnector``, or ``psycopg2``.
+- ``username``: The username to use when connecting to the database.
+- ``password``: The password to use when connecting to the database.
+- ``host``: The host to connect to.
+- ``port``: The port to connect to.
+- ``database``: The name of the database to create. For MySQL, this should be left blank. For PostgreSQL, this should be the name of schema in which the database will be created.
```

Comment:
Hmm, this does not parse. Postgres has databases and schemas inside databases. SQLAlchemy URLS specify database name as a first part of the path after host:port. So I'm confused is this is actually the database name in which a new schema will be created, or this specifies schema name and database name comes from Felis schema name (which sounds odd).

Reply:
I will clarify for since the syntax is different than with MySQL. You're right that target database has to be created beforehand and provided in the URL for Postgres.

Reply:
In MySQL, the database name is generally left out of the URL and it will come from the file automatically. Or it can be overridden using a CLI option.

For PostgreSQL, the database name must be included in the URL, and the schema is created within the target database, using the name within the schema or that from the CLI option.

I agree that it is a bit confusing since the URL is different depending on the target database variant (MySQL vs PostgreSQL). I do mention this several times in the documentation though.

Reply:
This is misworded for Postgres. I will fix it.